.github/workflows/instascale-release.yml

```diff
@@ -1,37 +1,73 @@
+# This workflow will release the Instascale repository: build an image for the instascale-controller, then push the image to the `project-codeflare` quay repo
 
 name: InstaScale Release
 
 on:
   workflow_dispatch:
-  push:
-    branches: [ "main" ]
-    paths:
-      - "VERSION"
+    inputs:
+      tag:
+        description: 'Release tag'
+        required: true
+        default: 'v0.0.0-dev'
+      mcad-version:
+        description: 'Published version of multi-cluster-app-dispatcher to be used as a dependency for InstaScale'
+        required: true
+        default: 'v0.0.0-dev'
+      release-title:
+        description: 'Title of the GitHub release'
+        required: true
+        default: ''
+      release-notes:
+        description: 'Text of the release notes'
+        required: true
+        default: ''
 
 jobs:
   release:
     runs-on: ubuntu-latest
+
+    # Permission required to create a release
+    permissions:
+      contents: write
+
     steps:
     - uses: actions/checkout@v3
 
-    - name: Grab New Version Tag
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.17
+
+    - name: Update to specific version of multi-cluster-app-dispatcher dependency
       run: |
-        echo "tag=$(cat VERSION)" >> $GITHUB_ENV
+        go get github.com/project-codeflare/multi-cluster-app-dispatcher@${{ github.event.inputs.mcad-version }}
 
+    # Build and test repository to make sure that MCAD update didn't break anything
+    - name: Build
+      run: go build -o bin/manager main.go
+    - name: Test
+      run: go test -v ./...
+
+    # Build and push the image
     - name: Login to Quay.io
       uses: redhat-actions/podman-login@v1
       with:
         username: ${{ secrets.QUAY_ID }}
         password: ${{ secrets.QUAY_TOKEN }}
         registry: quay.io
-
     - name: Image Build
       run: |
-        make image-build -e IMG=quay.io/project-codeflare/instascale-controller:v${{ env.tag }}
-        podman tag quay.io/project-codeflare/instascale-controller:v${{ env.tag }} quay.io/project-codeflare/instascale-controller:latest
-
+        make image-build -e IMG=quay.io/project-codeflare/instascale-controller:${{ github.event.inputs.tag }}
+        podman tag quay.io/project-codeflare/instascale-controller:${{ github.event.inputs.tag }} quay.io/project-codeflare/instascale-controller:latest
     - name: Image Push
       run: |
-        make image-push -e IMG=quay.io/project-codeflare/instascale-controller:v${{ env.tag }}
+        make image-push -e IMG=quay.io/project-codeflare/instascale-controller:${{ github.event.inputs.tag }}
         make image-push -e IMG=quay.io/project-codeflare/instascale-controller:latest
+
+    # Create a release in InstaScale
+    - run: |
+        gh release create ${{ github.event.inputs.tag }} --title "${{ github.event.inputs.release-title }}" --notes "${{ github.event.inputs.release-notes }}"
```

Comment:
It would be nice if the release somehow included a changelog -- a list of all the commits/issues closed or w/e between releases. We can think about further improving this later.

Reply:
Action step is adjusted, now the notes are generated automatically.
Notes contain a link to diff between last tag and current tag, see https://github.com/sutaakar/instascale/releases/tag/v0.0.3-dev